### PR TITLE
Computer player

### PR DESCRIPTION
**Pull request by Sillhouette**
This PR revolves around adding a new type of player, a computer player that can make random moves. It also abstracts some of the capabilities of the original player class into two seperate classes, human_player and computer_player

### CODE REVIEW

File: src/game.py [hunk 1]
```diff
@@ -22,8 +21,8 @@ def play(self):
         self.cli.print_board(self.board)
         while(self.game_in_process):
             current_player = self.current_player()
-            player_choice = self.cli.request_move(current_player)
-            selected_action, move = self.validator.validate(player_choice)
+            player_choice = current_player.get_move()
+            selected_action, move = player_choice
```
Comment (dcmoore) [inline src/game.py]: why have the player_choice variable? what does it add?

File: src/player_builder.py [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
+        self.cli = cli
+
     def build_players(self, player_tokens):
         players = []
-        for token in player_tokens:
-            players.append(Player(token=token))
+        players.append(HumanPlayer(self.cli, self.validator, player_tokens[0]))
+        players.append(ComputerPlayer(self.board, self.cli, player_tokens[1]))
+        #for token in player_tokens:
+        #    players.append(Player(cli, token))
```
Comment (dcmoore) [inline src/player_builder.py]: I think you forgot to delete these comments

**Review by dcmoore — APPROVED** [review]
Overall looks pretty good. Seems like these abstractions fit in with the game class nicely. Glad to see your architectural refactor start to pay off.

File: src/computer_player.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    def get_move(self):
+        self.cli.notify_for_computer_turn()
+
+        valid_space = False
+        
+        while valid_space == False:
+            choice = random.randint(1, 9)
+            index = self.board.move_to_index(choice)
+            if not self.board.position_taken(index):
```
Comment (tpetersen0308) [inline src/computer_player.py]: validating a computer move seems like it should be unnecessary

**Pull request merged**